src/mqt/bench/utils.py

```diff
@@ -248,46 +205,11 @@ def save_as_qasm(
         f.write(qc_str)
     f.close()
 
-    if gate_set == ["rz", "sx", "x", "ecr", "measure"]:
+    if gate_set == ["rz", "sx", "x", "ecr", "measure", "barrier"]:
```

Comment:
Could we substitute this list by some function that extracts the gates from the respective provider to avoid this duplicated code?